### ISSUE TITLE
Unable to find forwarding rule for host - Put Object on S3 Bucket

### ISSUE THREAD

**Issue by canpan14** [localstack/localstack#3175]
# Type of request

[X] bug report
[ ] feature request

# Detailed description
localstack verison - 0.12.1

I'm trying to make a call from a service running on a docker container to localstack which is running on it's own container.
The call does make it through to the localstack container but throws the following error:
Note: I changed the host to be localstack, but the problem persists no matter what the host is.

`Unable to find forwarding rule for host "localstack:4566", path "/test-bucket/foldername/filename", target header "", auth header "", data "b'filedatagoeshere'"`

I confirmed I was able to:
- Hit /health from inside the container, outside, and from the other service
- curl http://localhost:4566/fake-bucket and get an error for a invalid bucket name
- curl http://localhost:4566/test-bucket/fakefilename and get an error for an invalid key
- See the s3 bucket named test-bucket on the front end on localhost:8080
- curl -d 'data' -H 'Content-Type: text/plain' http://localhost:4566/test-bucket/fakefilename throws the same `Unable to find forwarding rule...` error

## Expected behavior
I expect a success of some sorts and the file is put on to the bucket

## Actual behavior
Service gets the error:
`(Service: Amazon S3; Status Code: 404; Error Code: 404 ; Request ID: null; S3 Extended Request ID: null)`
Localstack displays the error:
`Unable to find forwarding rule for host "localstack:4566", path "/test-bucket/foldername/filename", target header "", auth header "", data "b'filedatagoeshere'"`

# Steps to reproduce
1. Started up both containers
2. Call service from local computer
3. Service calls into localstack to put file in s3
4. localstack throws the reported error

## Command used to start LocalStack
Note: I use a mac, and there is other service that starts up as part of the docker-compose, just hiding that for this paste
```
version: "3.5"
services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack-full
    ports:
      - "4566-4599:4566-4599"
      - "8080:8080"
    environment:
      - HOSTNAME_EXTERNAL=localstack
      - HOSTNAME=localstack
      - SERVICES=s3
      - DEBUG=true
      - DATA_DIR=/tmp/localstack/data
      - PORT_WEB_UI=8080
      - DOCKER_HOST=unix:///var/run/docker.sock
      - TMPDIR=/private$TMPDIR
    volumes:
      - "/tmp/localstack:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
Code works fine when interacting with a real S3 bucket.
S3 Client Config
```
@Configuration
public class S3ClientConfig {
    private String url = "http://localstack:4566";
    private String awsRegion = "us-east-1";

    @Bean
    public AmazonS3 amazonS3() {
        return AmazonS3ClientBuilder.standard()
                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(url, awsRegion))
                .withPathStyleAccessEnabled(true)
                .build();
    }
}
```

```
        InputStream is = new ByteArrayInputStream(fileData);
        PutObjectRequest putObjectRequest = new PutObjectRequest("bucketName", "fileName", is, new ObjectMetadata());
        PutObjectResult result;
        try {
            result = amazonS3.putObject(putObjectRequest);
        } catch (SdkClientException e) {
            log.error("Encountered a SDKClientException when putting content onto AmazonS3" , e);
            throw e;
        }
```

**Comment by canpan14** [comment]
Additional debugging:

- I was able to use the aws cli to put a file on the s3 bucket `aws --debug s3 --endpoint-url=http://localhost:4566 cp re
quirements.txt s3://test-bucket/testfile.txt`
- I was able to get the following curl command through leading me to believe there must be something different from how the code is sending off the request compared to my curl command. Trouble is I'm just using the AWS SDK in a very basic way. I'll have to do more investigating.
```
curl -X PUT -d "data" -H "Content-Type: text/plain" http://localstack:4566/test-bucket/testfile.txt  -v
```

**Comment by canpan14** [comment]
I finally got it working.
Turns out localstack either doesn't support s3 folders or I need to do it a different way.
I was trying to upload to test-bucket/myfolder/myfile and was getting the error.

Simply taking off the folder and changing it to test-bucket/myfile allowed the routing to work.

@usmangani1 I'm going to close this, but let me know if there is a way of doing this that I'm missing.

**Comment by whummer** [comment]
Thanks for reporting @canpan14 . This should be fixed in #3500 - can you please pull the `latest` Docker image (currently being built by Travis-CI) and give it another try? Please report here if the problem persists. Thanks
